sql/postgresql/PostgreSQLParser.g4

```diff
@@ -3353,19 +3233,15 @@ opt_for_locking_clause
    ;
 
 for_locking_items
-   : for_locking_item
-   | for_locking_items for_locking_item
+   : for_locking_item+
    ;
 
 for_locking_item
    : for_locking_strength locked_rels_list opt_nowait_or_skip
    ;
 
 for_locking_strength
-   : FOR UPDATE
-   | FOR NO KEY UPDATE
-   | FOR SHARE
-   | FOR KEY SHARE
+   : FOR (UPDATE | NO KEY UPDATE | SHARE | KEY SHARE)
```

Comment:
```suggestion
   : FOR ((NO KEY)? UPDATE | KEY? SHARE)
```

Reply:
main idea was to avoid leading optionals values ( https://tomassetti.me/improving-the-performance-of-an-antlr-parser/)

Reply:
it would be good to proof that leading optional parts does not hurts performance/resource consumption, as it will allow to simplify grammar even more;

Reply:
I wrote the answer and awaiting a response in [the topic](https://groups.google.com/g/antlr-discussion/c/PpgPQU5jA3Q/m/gcsonPV2AgAJ).

> it would be good to proof that leading optional parts does not hurts performance/resource consumption

I'd vote for the opposite: proof the leading optional parts hurt performance. Otherwise consider they are not. Since they   make grammar dirtier and there is no proof that they affect performance I suggest using leading optionals where they clarify grammar.